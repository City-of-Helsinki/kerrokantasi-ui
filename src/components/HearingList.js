/* eslint-disable react/no-multi-comp */
import React from 'react';
import Helmet from 'react-helmet';
import PropTypes from 'prop-types';
import isEmpty from 'lodash/isEmpty';
import { Waypoint } from 'react-waypoint';
import { FormattedMessage, FormattedPlural, intlShape } from 'react-intl';
import { Nav, NavItem, FormGroup, FormControl, ControlLabel, Checkbox, Row, Col, Label } from 'react-bootstrap';
import { keys, capitalize } from 'lodash';

import FormatRelativeTime from '../utils/FormatRelativeTime';
import HearingsSearch from './HearingsSearch';
import Icon from '../utils/Icon';
import LabelList from './LabelList';
import Link from './LinkWithLang';
import LoadSpinner from './LoadSpinner';
import MouseOnlyLink from './MouseOnlyLink';
import OverviewMap from '../components/OverviewMap';
import config from '../config';
import getAttr from '../utils/getAttr';
import { labelShape } from '../types';
import { getHearingURL, isPublic } from '../utils/hearing';

// eslint-disable-next-line import/no-unresolved
import defaultImage from '@city-images/default-image.svg';

const HEARING_LIST_TABS = {
  LIST: 'list',
  MAP: 'map',
};

const HearingListTabs = ({activeTab, changeTab}) => (
  <Row>
    <Col md={8} mdPush={2}>
      <Nav className="hearing-list__tabs" bsStyle="tabs" activeKey={activeTab} role="tablist">
        {keys(HEARING_LIST_TABS).map(key => {
          const value = HEARING_LIST_TABS[key];
          return (
            <NavItem key={key} eventKey={value} title={capitalize(value)} onClick={() => changeTab(value)} role="tab">
              <FormattedMessage id={value} />
            </NavItem>
          );
        })}
      </Nav>
    </Col>
  </Row>
);

HearingListTabs.propTypes = {
  activeTab: PropTypes.string,
  changeTab: PropTypes.func,
};

class HearingListFilters extends React.Component {
  state = {
    sortChangeStatusMessages: [],
  }

  sortList = (event) => {
    const { handleSort } = this.props;

    handleSort(event.target.value);

    const newMessage = {
      id: Math.random(),
    };

    this.setState({ sortChangeStatusMessages: [newMessage] });

    // Clear the message after 5 seconds
    setTimeout(() => {
      this.setState({ sortChangeStatusMessages: [] });
    }, 5000);
  }

  render() {
    const { formatMessage } = this.props;

    return (
      <div className="hearing-list__filter-bar clearfix">
        <div className="sr-only">
          {this.state.sortChangeStatusMessages && this.state.sortChangeStatusMessages.map(alert => (
            <div key={alert.id} aria-live="assertive" role="status">
              <FormattedMessage id="orderHasBeenChanged" />
            </div>
          ))}
        </div>
        <FormGroup controlId="formControlsSelect" className="hearing-list__filter-bar-filter">
          <ControlLabel className="hearing-list__filter-bar-label">
            <FormattedMessage id="sort" />
          </ControlLabel>
          <FormControl
            className="select"
            componentClass="select"
            placeholder="select"
            onChange={event => this.sortList(event)}
          >
            <option value="-created_at">{formatMessage({id: 'newestFirst'})}</option>
            <option value="created_at">{formatMessage({id: 'oldestFirst'})}</option>
            <option value="-close_at">{formatMessage({id: 'lastClosing'})}</option>
            <option value="close_at">{formatMessage({id: 'firstClosing'})}</option>
            <option value="-open_at">{formatMessage({id: 'lastOpen'})}</option>
            <option value="open_at">{formatMessage({id: 'firstOpen'})}</option>
            <option value="-n_comments">{formatMessage({id: 'mostCommented'})}</option>
            <option value="n_comments">{formatMessage({id: 'leastCommented'})}</option>
          </FormControl>
        </FormGroup>
      </div>
    );
  }
}

HearingListFilters.propTypes = {
  handleSort: PropTypes.func,
  formatMessage: PropTypes.func,
};

export class HearingListItem extends React.Component {
  render() {
    const { hearing, language, history, formatTime, formatDate } = this.props;
    const mainImage = hearing.main_image;
    let mainImageStyle = {
      backgroundImage: `url(${defaultImage})`,
    };
    if (hearing.main_image) {
      mainImageStyle = {
        backgroundImage: 'url("' + mainImage.url + '")',
      };
    }

    const translationAvailable = !!getAttr(hearing.title, language, { exact: true });
    const availableInLanguageMessages = {
      fi: 'Kuuleminen saatavilla suomeksi',
      sv: 'Hörandet tillgängligt på svenska',
      en: 'Questionnaire available in English',
    };

    return (
      <div className="hearing-list-item" role="listitem">
        <MouseOnlyLink
          className="hearing-list-item-image"
          style={mainImageStyle}
          history={history}
          url={getHearingURL(hearing)}
          altText={getAttr(hearing.title, language)}
        />
        <div className="hearing-list-item-content">
          <div className="hearing-list-item-title-wrap">
            <h2 className="h4 hearing-list-item-title">
              <Link to={{ path: getHearingURL(hearing) }}>
                {!isPublic(hearing) ? (
                  <FormattedMessage id="hearingListNotPublished">
                    {(label) => <Icon name="eye-slash" aria-label={label} />}
                  </FormattedMessage>
                ) : null}{' '}
                {getAttr(hearing.title, language)}
              </Link>
            </h2>
            <div className="hearing-list-item-comments">
              <Icon name="comment-o" aria-hidden="true" />&nbsp;{hearing.n_comments}
              <span className="sr-only">
                {hearing.n_comments === 1 ? (
                  <FormattedMessage id="hearingListComment" />
                ) : <FormattedMessage id="hearingListComments" />}
              </span>
            </div>
          </div>
          <div className="hearing-list-item-times">
            <div>
              <FormatRelativeTime messagePrefix="timeOpen" timeVal={hearing.open_at} />
            </div>
            <div>
              <FormatRelativeTime
                messagePrefix="timeClose"
                timeVal={hearing.close_at}
                formatTime={formatTime}
                formatDate={formatDate}
              />
            </div>
          </div>
          <div className="hearing-list-item-labels clearfix">
            <LabelList labels={hearing.labels} className="hearing-list-item-labellist" language={language} />
            {hearing.closed ? (
              <div className="hearing-list-item-closed">
                <Label>
                  <FormattedMessage id="hearingClosed" />
                </Label>
              </div>
            ) : null}
          </div>
          {!translationAvailable && (
            <div className="hearing-card-notice">
              <Icon name="exclamation-circle" aria-hidden="true" />
              <FormattedMessage id="hearingTranslationNotAvailable" />
              {config.languages.map(
                lang =>
                  (getAttr(hearing.title, lang, { exact: true }) ? (
                    <div key={lang} className="language-available-message">
                      {availableInLanguageMessages[lang]}
                    </div>
                  ) : null)
              )}
            </div>
          )}
        </div>
      </div>
    );
  }
}

HearingListItem.propTypes = {
  hearing: PropTypes.object,
  language: PropTypes.string,
  history: PropTypes.object,
  formatTime: PropTypes.func,
  formatDate: PropTypes.func,
};

export const HearingList = ({
  handleSearch,
  handleSelectLabels,
  handleSort,
  hearings,
  hearingCount,
  isLoading,
  isMobile,
  labels,
  language,
  onTabChange,
  searchPhrase,
  selectedLabels,
  showOnlyOpen,
  tab: activeTab,
  toggleShowOnlyOpen,
  handleReachBottom,
  intl
}) => {
  const hearingsToShow = !showOnlyOpen ? hearings : hearings.filter(hearing => !hearing.closed);
  const hasHearings = !isEmpty(hearings);
  const {formatMessage, formatTime, formatDate} = intl;

  const hearingListMap = hearingsToShow ? (
    <Row>
      <Col xs={12}>
        <Helmet title={formatMessage({ id: 'mapView' })} />
        <div className="hearing-list-map map">
          <Checkbox inline readOnly checked={showOnlyOpen} onChange={toggleShowOnlyOpen} style={{marginBottom: 10}}>
            <FormattedMessage id="showOnlyOpen" />
          </Checkbox>
          <OverviewMap
            hearings={hearingsToShow}
            style={{width: '100%', height: isMobile ? '100%' : 600}}
            enablePopups
          />
        </div>
      </Col>
    </Row>
  ) : null;

  return (
    <div>
      <section className="page-section--hearings-search">
        <div className="container">
          <Row>
            <Col md={10} mdPush={1}>
              <HearingsSearch
                handleSearch={handleSearch}
                handleSelectLabels={handleSelectLabels}
                labels={labels}
                language={language}
                searchPhrase={searchPhrase}
                selectedLabels={selectedLabels}
                intl={intl}
              />
            </Col>
          </Row>
        </div>
      </section>
      <section className="page-section--hearings-tabs">
        <div className="container">
          <HearingListTabs activeTab={activeTab} changeTab={onTabChange} />
        </div>
      </section>
      <section className="page-section page-section--hearings-list" id="hearings-section">
        <a href="#hearings-search-form" className="sr-only">
          <FormattedMessage id="jumpToSearchForm" />
        </a>
        <div className="container">
          <Row>
            <Col md={8} mdPush={2}>
              {(!isLoading && !hasHearings) && (
                <h2 className="hearing-list__hearing-list-title">
                  <FormattedMessage id="noHearings">{txt => txt}</FormattedMessage>
                </h2>
              )}

              {(hasHearings && activeTab === 'list') && (
                <div>
                  <div className="hearing-list__result-controls">
                    <h2 className="hearing-list__hearing-list-title">
                      <FormattedPlural
                        value={hearingCount}
                        one={
                          <FormattedMessage id="totalNumHearing" values={{ n: hearingCount }}>
                            {txt => txt}
                          </FormattedMessage>
                        }
                        other={
                          <FormattedMessage id="totalNumHearings" values={{ n: hearingCount }}>
                            {txt => txt}
                          </FormattedMessage>
                        }
                      >{txt => txt}
                      </FormattedPlural>
                    </h2>
                    <HearingListFilters handleSort={handleSort} formatMessage={formatMessage} />
                  </div>

                  <div className="hearing-list">
                    <div role="list">
                      {hearings.map(hearing => (
                        <HearingListItem
                          hearing={hearing}
                          key={hearing.id}
                          language={language}
                          formatTime={formatTime}
                          formatDate={formatDate}
                        />
                      ))}
                    </div>
                    {isLoading && <LoadSpinner />}
                    {!isLoading && <Waypoint onEnter={handleReachBottom} />}
                  </div>
                </div>
              )}
            </Col>
          </Row>
          {hasHearings && activeTab === 'map' && !isLoading ? hearingListMap : null}
        </div>
      </section>
    </div>
  );
};

HearingList.propTypes = {
  handleSearch: PropTypes.func,
  handleSelectLabels: PropTypes.func,
  handleSort: PropTypes.func,
  hearings: PropTypes.array,
  hearingCount: PropTypes.number,
  intl: intlShape.isRequired,
  isLoading: PropTypes.bool,
  isMobile: PropTypes.bool,
  labels: PropTypes.arrayOf(labelShape),
  language: PropTypes.string,
  onTabChange: PropTypes.func,
  searchPhrase: PropTypes.string,
  selectedLabels: PropTypes.arrayOf(PropTypes.string),
  showOnlyOpen: PropTypes.bool,
  tab: PropTypes.string,
  toggleShowOnlyOpen: PropTypes.func,
  handleReachBottom: PropTypes.func,
};

HearingList.defaultProps = {
  tab: HEARING_LIST_TABS.LIST,
  hearingCount: 0,
};

export default HearingList;
