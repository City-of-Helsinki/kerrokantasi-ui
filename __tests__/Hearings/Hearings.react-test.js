import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {Hearings} from '../../src/views/Hearings/index';
import {HearingList} from '../../src/components/HearingList';
import {MemoryRouter} from 'react-router-dom';
import {mockStore, getIntlAsProp} from '../../test-utils';

// Renders the Hearings component using enzymes shallow rendering.
// You can pass props you want to override as a parameter.
const setup = propOverrides => {
  const {labels, ...rest} = mockStore;
  const props = Object.assign({
    labels: labels.data,
    ...rest
  }, propOverrides);

  const wrapper = shallow(<MemoryRouter><Hearings intl={getIntlAsProp()} {...props} /></MemoryRouter>);

  return {
    props,
    wrapper
  };
};

test('Hearings component should render as expected', () => {
  const {wrapper} = setup();
  const tree = toJson(wrapper.dive().dive());
  expect(tree).toMatchSnapshot();
});

test('Hearings component will render HearingList when labels are present', () => {
  const {wrapper} = setup();
  expect(wrapper.dive().dive().find(HearingList)).toHaveLength(1);
});
