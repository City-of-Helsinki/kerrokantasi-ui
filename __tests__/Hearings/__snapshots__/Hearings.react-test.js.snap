// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hearings component should render as expected 1`] = `
<div
  className="hearings"
>
  <section
    className="page-section page-section--all-hearings-header"
  >
    <div
      className="container"
    >
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          md={10}
          mdPush={1}
        >
          <HelmetWrapper
            title="Kaikki kuulemiset"
          />
          <h1
            className="page-title"
          >
            <FormattedMessage
              id="allHearings"
              tagName="span"
              values={Object {}}
            />
          </h1>
        </Col>
      </Row>
    </div>
  </section>
  <HearingList
    handleReachBottom={[Function]}
    handleSearch={[Function]}
    handleSelectLabels={[Function]}
    handleSort={[Function]}
    hearings={
      Array [
        Object {
          "abstract": Object {
            "fi": "Tämä on suomenkielinen testiabstrakti. Sisältää huikean kuvauksen kyseisestä kuulemisesta.",
          },
          "borough": Object {},
          "close_at": "2200-11-28T22:00:00Z",
          "closed": false,
          "created_at": "2017-10-23T11:07:26.630423Z",
          "default_to_fullscreen": false,
          "geojson": null,
          "id": "rXT2L1HxEOZTERjluyxyQZ412aYM8oZE",
          "labels": Array [],
          "main_image": Object {
            "caption": Object {
              "fi": "A Wonderfull caption",
            },
            "height": 631,
            "id": 92,
            "title": Object {},
            "url": "https://api.hel.fi/kerrokantasi-test/media/images/2017/10/LzqVMrMI.jpeg",
            "width": 893,
          },
          "n_comments": 1,
          "open_at": "2017-10-22T21:00:00Z",
          "organization": "Mock Inc",
          "published": true,
          "servicemap_url": "",
          "slug": "mockHearing",
          "title": Object {
            "fi": "Mock Hearing",
          },
        },
      ]
    }
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "locale": "fi",
        "messages": Object {
          "APICallFailed": "Jokin meni pieleen. Kokeile pian uudestaan tai päivitä selain.",
          "CREATED_AT_ASC": "Vanhin ensin",
          "CREATED_AT_DESC": "Uusin ensin",
          "POPULARITY_ASC": "Vähiten suosittu ensin",
          "POPULARITY_DESC": "Suosituin ensin",
          "addComment": "Vastaa",
          "addGeojson": "Lisää geojson tiedostona",
          "addOption": "Lisää vastausvaihtoehto",
          "addProcess": "Lisää uusi vaihe",
          "addSection": "Lisää osa-alue",
          "add_images": "Lisää kuvia kommenttiisi",
          "all": "Kaikki",
          "allHearings": "Kaikki kuulemiset",
          "alreadyVoted": "Olet jo antanut äänesi tälle kommentille.",
          "anonymous": "Anonyymi",
          "asOpenData": "avoimena datana",
          "borough": "Alue",
          "cancel": "Peru",
          "characters": "merkkiä",
          "chooseTopic": "Valitse aihe",
          "choose_images": "Valitse kuvia",
          "closeHearing": "Sulje kuuleminen",
          "closedHearing": "Suljettu",
          "commentConditions": "Annetut kommentit julkaistaan {linkToDefinition} Helsingin kaupungin palvelussa {linkToLicense}. Voit jättää kommenttisi nimettömänä, jos et ole kirjautunut sisään. Lisäksi tekstejä tarvittaessa muokataan loukkaavan tai muutoin törkeän kielenkäytön osalta.",
          "commentDeleted": "Kommenttisi on poistettu.",
          "commentEdited": "Kommenttisi on muokattu. Kiitos!",
          "commentOrder": "Järjestä kommentit:",
          "commentReceived": "Kommenttisi on vastaanotettu. Kiitos!",
          "comments": "Mielipiteet",
          "commentsOnMap": "Kartta koko näytöllä",
          "commentsOutOf": "Näytetään {current}/{total} kommenttia.",
          "contactPersons": "Yhteyshenkilöt",
          "contactTitle": "Virkanimike tai tehtäväkuvaus",
          "contactTitlePlaceholder": "Anna yhteyshenkilön virkanimike tai tehtävänkuvaus...",
          "create": "Luo",
          "createContact": "Luo uusi yhteyshenkilö",
          "createHearing": "Luo uusi kuuleminen",
          "createLabel": "Luo uusi asiasana",
          "defaultClosureInfo": "Tämä kuuleminen on sulkeutunut. Mielipiteiden antaminen ja kannattaminen ei enää ole mahdollista.",
          "defaultProject": "Lisää uusi projekti",
          "delete": "Poista",
          "deleteComment": "Poista kommentti",
          "deleteConfirmation": "Oletko varma, että haluat poistaa kommentin?",
          "deleteDraft": "Poista luonnos",
          "deleteQuestion": "Poista kysymys",
          "deleteSection": "Poista osio",
          "downloadReport": "Lataa yhteenveto kommenteista Excel-muodossa",
          "draft": "Luonnos",
          "draftDeleted": "Luonnos poistettiin onnistuneesti",
          "draftNotPublished": "Luonnos, ei julkaistu",
          "drafts": "Luonnokset",
          "edit": "Muokkaa",
          "editContact": "Muokkaa yhteystietoa",
          "editHearing": "Muokkaa",
          "email": "Sähköposti",
          "eyeTooltip": "Tätä kuulemista ei ole julkaistu eikä se ole julkisesti nähtävissä.",
          "eyeTooltipOpens": "Tämä kuuleminen julkaistaan",
          "eyeTooltipOpensDays": "päivän kuluttua. Se ei ole vielä julkisesti nähtävissä.",
          "eyeTooltipOpensHours": "tunnin kuluttua. Se ei ole vielä julkisesti nähtävissä.",
          "feedbackPrompt": "Kerro meille, miten tätä palvelua pitäisi kehittää!",
          "firstClosing": "Ensimmäisenä sulkeutuva",
          "firstOpen": "Ensimmäisenä auennut",
          "follow": "Seuraa",
          "followingHearing": "Seuraat kuulemista.",
          "furtherInformation": "Lisätietoja",
          "hearing": "Kuuleminen",
          "hearingAbstract": "Lyhyt johdanto",
          "hearingArea": "Jos kuuleminen ei koske koko kaupunkia, piirrä kuulemisen tarkka alue kartalle",
          "hearingChangesSaved": "Muutokset tallennettu",
          "hearingClosed": "Sulkeutunut",
          "hearingClosingTime": "Sulkeutumisaika",
          "hearingClosingTimePlaceholder": "Valitse...",
          "hearingClosureInfo": "Sulkeutumistiedote",
          "hearingCommenting": "Kommentointi",
          "hearingContacts": "Yhteyshenkilöt",
          "hearingContactsPlaceholder": "Valitse kuulemisen yhteyshenkilöt",
          "hearingFormHeaderStep1": "1. Kuulemisen perustiedot",
          "hearingFormHeaderStep2": "2. Sisältö",
          "hearingFormHeaderStep3": "3. Kartta",
          "hearingFormHeaderStep4": "4. Julkaisutiedot",
          "hearingFormHeaderStep5": "5. Hankkeen vaiheet",
          "hearingFormNext": "Jatka",
          "hearingImage": "Kuulemisen kuva",
          "hearingLabels": "Asiasanat",
          "hearingLabelsPlaceholder": "Valitse kuulemiselle sopivat asiasanat",
          "hearingLanguages": "Kuulemisen tiedot syötetään",
          "hearingMainContent": "Kuulemisen esittelyteksti",
          "hearingMap": "Kuulemiset kartalla",
          "hearingMapHeaderDescription": "Näe kuulemiset kartalla",
          "hearingMapHeaderText": "Kartta",
          "hearingNotFound": "Hakemaasi kuulemista ei löytynyt.",
          "hearingOpeningTime": "Avautumisaika",
          "hearingOpeningTimePlaceholder": "Valitse...",
          "hearingSavingFailed": "Tallentaminen epäonnistui",
          "hearingSlug": "Kuulemisen osoite (täytä osoitteen loppuosa)",
          "hearingSlugPlaceholder": "esimerkki-osoite",
          "hearingTranslationNotAvailable": "Tätä kuulemista ei ole julkaistu suomeksi",
          "hearings": "Kuulemiset",
          "hearingsHeaderDescription": "Antamalla kommenttisi voit vaikuttaa käsiteltävinä olevien asioiden valmisteluun",
          "hearingsHeaderText": "Kuulemiset",
          "help": "Ohjeet",
          "imageSizeError": "Kuva liian suuri, maksimikoko 1MB.",
          "image_too_big": "Kuva liian suuri, kuvan suurin sallitu koko on 1MB",
          "inLanguage-en": "englanniksi",
          "inLanguage-fi": "suomeksi",
          "inLanguage-sv": "ruotsiksi",
          "infoHeaderDescription": "Tämä on palvelun väliaikainen prototyyppi, josta toivomme käyttäjien kommentteja",
          "infoHeaderText": "Tietoa palvelusta",
          "infoPage": "Tietoa palvelusta",
          "labelPlaceholder": "Anna asiasanan nimi...",
          "lang-en": "English",
          "lang-fi": "Suomi",
          "lang-sv": "Svenska",
          "lastClosing": "Viimeisenä sulkeutuva",
          "lastOpen": "Viimeisenä auennut",
          "leastCommented": "Vähiten kommentoitu",
          "list": "Lista",
          "logInToAnswer": "Kirjaudu sisään äänestääksesi ja katsoaksesi tuloksia",
          "login": "Kirjaudu sisään",
          "loginToComment": "Kirjaudu sisään kommentoidaksesi.",
          "loginToContinue": "Kirjaudu sisään jatkaaksesi",
          "loginToVoteComment": "Kirjaudu sisään äänestääksesi kommenttia.",
          "logout": "Kirjaudu ulos",
          "mainSection": "Pääosio",
          "map": "Kartta",
          "mapView": "Kaikki kuulemiset kartalla",
          "mostCommented": "Eniten kommentoitu",
          "moveDown": "Siirrä alas",
          "moveUp": "Siirrä ylös",
          "multipleChoiceQuestion": "Monivalintakysymys (vastaaja voi valita useamman vaihtoehdon)",
          "multipleImages": "Voit lisätä useita kuvia klikkaamalla useampaa tiedostoa samalla kun pidät Ctrl- tai komentonäppäintä painettuna.",
          "name": "Nimi",
          "newHearingPageTitle": "Uusi kuuleminen",
          "newMultipleChoiceQuestion": "Lisää monivalintakysymys",
          "newSingleChoiceQuestion": "Lisää yksivalintakysymys",
          "newestFirst": "Uusin ensin",
          "newestHearings": "Uusimmat kuulemiset",
          "next": "Seuraava",
          "nextClosingHearing": "Seuraavaksi sulkeutuva kuuleminen",
          "nickname": "Nimimerkki",
          "noCommenting": "Ei kommentointia",
          "noCommentsAvailable": "Ei kommentteja",
          "noHearings": "Kuulemisia ei löytynyt",
          "noProject": "Ei projektia",
          "oldestFirst": "Vanhin ensin",
          "open-hearings-on-map": "Käynnissä olevat kuulemiset kartalla",
          "openCommenting": "Avoin kommentointi",
          "openFullscreenMap": "Kartta koko näytöllä",
          "openHearings": "Avoimet kuulemiset",
          "option": "Vastausvaihtoehto",
          "overview-map": "Kartta",
          "partSection": "Osa-alue",
          "phase": "Vaihe",
          "phaseActive": "Kuuleminen kuuluu tähän vaiheeseen.",
          "phaseDescription": "Vaiheen kuvaus",
          "phaseDuration": "Vaiheen kesto",
          "phone": "Puhelinnumero",
          "previous": "Edellinen",
          "privacyPolicy": "Rekisteriseloste",
          "project": "Hanke:",
          "projectName": "Projektin nimi",
          "projectSelection": "Jos kuuleminen kuuluu monivaiheiseen projektiin, valitse projekti tästä.",
          "publishHearing": "Hyväksy julkaistavaksi",
          "publishHearingNow": "Julkaise kuuleminen",
          "published": "Julkaistu",
          "publishedHearings": "Julkiset",
          "publishingQueue": "Julkaisujono",
          "question": "Kysymys",
          "questionOptionPlaceholder": "Vastausvaihtoehdon teksti.",
          "questionTextPlaceholder": "Kirjoita tähän kysymys, johon haluat kaupunkilaisten vastaavan.",
          "register": "Rekisteröidy",
          "registered": "rekisteröitynyt",
          "registeredUsersOnly": "Vain kirjautuneet käyttäjät",
          "removeImage": "Poista kuva",
          "revertPublishing": "Peru julkaisu",
          "save": "Tallenna",
          "saveAndPreviewHearing": "Tallenna ja tarkastele luonnosta",
          "saveFailed": "Tallentaminen epäonnistui",
          "saveHearingChanges": "Tallenna muutokset",
          "scenarioSection": "Vaihtoehto",
          "search": "Etsi",
          "searchLabels": "Hae aiheista",
          "searchPlaceholder": "Etsi...",
          "searchSubjects": "Etsi aiheista",
          "searchTitles": "Etsi otsikoista",
          "sectionAbstract": "Lyhyt johdanto",
          "sectionAbstractPlaceholder": "Kuvaa tässä pääasia lyhyesti. Näytetään alussa suuremmalla kirjasinkoolla kuin muu tekstisisältö.",
          "sectionContent": "Tekstisisältö",
          "sectionContentPlaceholder": "Varsinainen tekstisisältö, johon voit lisätä muotoiluja, linkkejä ja kuvia",
          "sectionImage": "Kuva",
          "sectionImageCaption": "Kuvateksti",
          "sectionImageHelpText": "Kuvan suosituskoko on 960 × 720. (Max. 1MB)",
          "sectionImagePlaceholder": "Selitä tai täydennä kuvaa kuvatekstillä. Mainitse kuvaajan nimi, jos tarpeellista.",
          "sectionTitle": "Osa-alueen otsikko",
          "sectionTitlePlaceholder": "Lisää osa-alueelle kuvaava otsikko.",
          "selectAnotherImage": "Valitse toinen kuva",
          "selectOrDropImage": "Valitse tai pudota kuva",
          "showOnlyOpen": "Näytetään vain avoimet",
          "singleChoiceQuestion": "Yksivalintakysymys (vastaaja valitsee yhden vaihtoehdon)",
          "slugInUse": "Valitsemasi osoite oli käytössä ja siksi vaihdettiin.",
          "sort": "Järjestä",
          "submit": "Lähetä",
          "table-of-content": "Sisällys",
          "timeCloseFuture": "Sulkeutuu",
          "timeClosePast": "Sulkeutui",
          "timeOpenFuture": "Avautuu",
          "timeOpenPast": "Avautui",
          "timetable": "Aikajana",
          "title": "Otsikko",
          "titlePlaceholder": "Anna kuulemiselle kuvaava otsikko...",
          "toBePublishedHearing": "Kuuleminen julkaistaan",
          "totalSubmittedComment": "Yhteensä {n} kommentti",
          "totalSubmittedComments": "Yhteensä {n} kommenttia",
          "tryToFixFormErrors": "Yritä korjata seuraavat ongelmat ja kokeile uudestaan",
          "tryingToDeletePhaseWithHearings": "Vaihetta, jossa on jo kuulemisia, ei voida poistaa.",
          "vote": "Äänestä",
          "voteReceived": "Ääni vastaanotettu. Kiitos!",
          "welcome": "Kerro mielipiteesi",
          "welcomeMessage": "Tule mukaan kehittämään kaupunkia ja osallistu avoimiin kuulemisiin.",
          "withOpenLicense": "avoimella lisenssillä",
          "writeComment": "Kommentti",
        },
        "now": [Function],
      }
    }
    isLoading={true}
    labels={
      Array [
        Object {
          "id": 1,
          "label": Object {
            "fi": "Mock Von Label",
          },
        },
        Object {
          "id": 2,
          "label": Object {
            "fi": "Mock Don Label",
          },
        },
      ]
    }
    onTabChange={[Function]}
    selectedLabels={Array []}
    showOnlyOpen={false}
    tab="list"
    toggleShowOnlyOpen={[Function]}
  />
</div>
`;
